src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -1,40 +1,142 @@
 package edu.kis.vh.nursery.list;
 
 public class IntLinkedList {
+	
+	private class Node {
 
-	Node last;
-	int i;
+		private int value;
+		private Node prev;
+		private Node next;
 
+		/**
+		 * Konstruktor ustawiajacy wartosc wezla
+		 * @param i nowa wartosc wezla
+		 */
+		public Node(int i) {
+			setValue(i);
+		}
+
+		/**
+		 * Metoda umozliwiajaca pobranie wartosci wezla
+		 * @return wartosc wezla
+		 */
+		public int getValue() {
+			return value;
+		}
+
+		/**
+		 * Metoda umozliwiajaca ustawienie wartosci wezla
+		 * @param value nowa wartosc wezla
+		 */
+		public void setValue(int value) {
+			this.value = value;
+		}
+
+		/**
+		 * Pobranie poprzedniego wezla
+		 * @return porzedni wezel
+		 */
+		public Node getPrev() {
+			return prev;
+		}
+
+		/**
+		 * Ustawienie poprzedniego wezla
+		 * @param prev nowy wezel
+		 */
+		public void setPrev(Node prev) {
+			this.prev = prev;
+		}
+
+		/**
+		 * Pobranie nastepnego wezla
+		 * @return nastepny wezel
+		 */
+		public Node getNext() {
+			return next;
+		}
+
+		/**
+		 * Ustawienie nastepnego wezla
+		 * @param next nowy wezel
+		 */
+		public void setNext(Node next) {
+			this.next = next;
+		}
+	}
+
+
+	private Node last;
+	private int i;
+
+	/**
+	 * Metoda umozliwiajaca wstawienie nowego wezla do listy
+	 * @param i wartosc nowego wezla
+	 */
 	public void push(int i) {
-		if (last == null)
-			last = new Node(i);
+		if (getLast() == null) {
+			setLast(new Node(i));
+		}
 		else {
-			last.next = new Node(i);
-			last.next.prev = last;
-			last = last.next;
+			getLast().setNext(new Node(i));
+			getLast().getNext().setPrev(getLast());
+			setLast(getLast().getNext());
```

Comment:
This is an exaggeration. Using private fields internally (last field) usually does not require encapsulation.